notanorm/base.py

```diff
@@ -998,8 +1008,15 @@ def __select_to_query(
         if _limit is not None:
             sql += " " + self.limit_query(_limit)
 
+        if _group_by is not None:
```

Comment:
At least for SQLite, `GROUP BY` has to come before `ORDER BY` and `LIMIT` -- see diagram at https://www.sqlite.org/lang_select.html.

Reply:
right, thanks, will add test cases